assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Pagination.js

```diff
@@ -26,20 +26,52 @@ import { Icon, chevronLeft, chevronRight } from '@wordpress/icons';
  * WordPress dependencies
  */
 import { _x, sprintf } from '@wordpress/i18n';
+import { useCallback } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import Button from '../../../../../components/Button';
-import { IDEA_HUB_IDEAS_PER_PAGE } from '../../../datastore/constants';
+import {
+	IDEA_HUB_IDEAS_PER_PAGE,
+	MODULES_IDEA_HUB,
+} from '../../../datastore/constants';
+import { CORE_UI } from '../../../../../googlesitekit/datastore/ui/constants';
+import Data from 'googlesitekit-data';
+
+const { useSelect, useDispatch } = Data;
+
+const Pagination = ( { tab } ) => {
+	const uniqueKey = `idea-hub-page-${ tab }`;
+	const page =
+		useSelect( ( select ) => select( CORE_UI ).getValue( uniqueKey ) ) || 1;
+
+	const total = useSelect( ( select ) => {
+		if ( tab === 'new-ideas' ) {
+			return select( MODULES_IDEA_HUB ).getNewIdeas()?.length || 0;
+		}
+		if ( tab === 'saved-ideas' ) {
+			return select( MODULES_IDEA_HUB ).getSavedIdeas()?.length || 0;
+		}
+		if ( tab === 'draft-ideas' ) {
+			return select( MODULES_IDEA_HUB ).getDraftPostIdeas()?.length || 0;
+		}
+	} );
+
+	const { setValue } = useDispatch( CORE_UI );
+
+	const handlePrev = useCallback( () => {
+		if ( page > 1 ) {
+			setValue( uniqueKey, page - 1 );
+		}
+	}, [ page, setValue, uniqueKey ] );
+
+	const handleNext = useCallback( () => {
+		if ( page < Math.ceil( total / IDEA_HUB_IDEAS_PER_PAGE ) ) {
+			setValue( uniqueKey, page + 1 );
+		}
+	}, [ page, setValue, total, uniqueKey ] );
```

Comment:
Let's not display the pagination if there are no ideas:

```suggestion
	}, [ page, setValue, total, uniqueKey ] );

	if ( total < 1 ) {
		return null;
	}
```